src/Common/Commands.Common.Test/TestPowerBICmdletInitFactory.cs

```diff
@@ -29,14 +26,15 @@ public TestLogger Logger
             }
         }
 
-        public TestPowerBICmdletInitFactory(FakeHttpClientHandler clientHandler) : 
-            base(new TestLoggerFactory(), new ModuleDataStorage(), new TestAuthenticator(), new PowerBISettings(), new TestClient(clientHandler))
-        {
-        }
+        public TestPowerBICmdletInitFactory(IPowerBIApiClient client) :
+            base(new TestLoggerFactory(), new ModuleDataStorage(), new TestAuthenticator(), new PowerBISettings(), new TestClient(client)) => this.SetProfile();
+
+        public TestPowerBICmdletInitFactory(FakeHttpClientHandler clientHandler) :
+            base(new TestLoggerFactory(), new ModuleDataStorage(), new TestAuthenticator(), new PowerBISettings(), new TestClient(clientHandler)) => this.SetProfile();
 
-        public void SetProfile(IPowerBIProfile profile = null)
```

Comment:
I would keep that public so someone could reset it to use a different test profile